tests/functional/ssl/__init__.py

```diff
@@ -0,0 +1,118 @@
+# coding=utf-8
+# -----------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -----------------------------------------------------------------------------
+
+"""Setup for the package tests."""
+
+import multiprocessing
+import os
+import shutil
+import subprocess
+import time
+
+from libtest import project
+from libtest import codechecker
+from libtest import env
+
+# Stopping event for CodeChecker server.
+__STOP_SERVER = multiprocessing.Event()
+
+# Test workspace initialized at setup for authentication tests.
+TEST_WORKSPACE = None
+
+
+def setup_package():
+    """Setup the environment for the tests then start the server."""
+
+    global TEST_WORKSPACE
+    TEST_WORKSPACE = env.get_workspace('ssl')
+
+    os.environ['TEST_WORKSPACE'] = TEST_WORKSPACE
+
+    test_config = {}
+
+    # Setup environment variables for the test cases.
+    host_port_cfg = {'viewer_host': 'localhost',
+                     'viewer_port': env.get_free_port(),
+                     'viewer_product': 'authentication'}
+
+    test_env = env.test_env(TEST_WORKSPACE)
+
+    codechecker_cfg = {
+        'check_env': test_env,
+        'workspace': TEST_WORKSPACE,
+        'checkers': []
+    }
+
+    codechecker_cfg.update(host_port_cfg)
+
+    codechecker_cfg['run_names'] = []
+
+    test_config['codechecker_cfg'] = codechecker_cfg
+
+    # Export configuration for the tests.
+    env.export_test_cfg(TEST_WORKSPACE, test_config)
+
+    # Enable SSL
+    env.enable_ssl(TEST_WORKSPACE)
+
+    # Enable authentication and start the CodeChecker server.
+    env.enable_auth(TEST_WORKSPACE)
+    print("Starting server to get results")
+    _start_server(codechecker_cfg, test_config, False)
+
+
+def teardown_package():
+    """Stop the CodeChecker server and clean up after the tests."""
+    # TODO If environment variable is set keep the workspace
+    # and print out the path.
+    global TEST_WORKSPACE
+
+    # Removing the product through this server requires credentials.
+    codechecker_cfg = env.import_test_cfg(TEST_WORKSPACE)['codechecker_cfg']
+    codechecker.remove_test_package_product(TEST_WORKSPACE,
```

Comment:
If you didn't call `add_test_package_product()` in `startUp()`, you should not (need not) call `remove_test_package_product` here.